insights/client/connection.py

```diff
@@ -676,6 +676,28 @@ def unregister(self):
             logger.error("Could not unregister this system")
             return False
 
+    def unregister(self):
+        """
+        Unregister this system from the insights service
+        """
+        if self.config.legacy_upload:
+            return self._legacy_unregister()
+
+        results = self._fetch_system_by_machine_id()
+        msg_name = determine_hostname(self.config.display_name)
+        try:
+            logger.debug("Unregistering %s", msg_name)
+            url = self.api_url + "/inventory/v1/hosts/" + results[0]['id']
+            net_logger.info("DELETE %s", url)
+            self.session.delete(url)
+            logger.info(
+                "Successfully unregistered from the Red Hat Insights Service")
+            return True
+        except requests.ConnectionError as e:
```

Comment:
We may want to expand this to `(requests.ConnectionError, requests.Timeout)` based on some other feedback we've been getting about request exception stuff.

Reply:
:+1: